canteenWeb/models.py

```diff
@@ -7,15 +7,35 @@
 
 
 class User(AbstractUser):
-    pass
+    USERNAME_FIELD = models.CharField(max_length=30, unique=True, default="")  # sap id
+    EMAIL_FIELD = models.EmailField(max_length=40, blank=True, unique=True)
+    first_name = models.CharField(max_length=15)
+    last_name = models.CharField(max_length=15)
+    phone_number = models.CharField(max_length=12, blank=True)
+    device_id = models.CharField(max_length=30, default="")  # For use by android team
+    is_student = models.BooleanField(
+        default=True
+    )  # to check whether the user is student
+    is_teacher = models.BooleanField(
+        default=False
+    )  # to check whether the user is teacher
+    is_banned = models.BooleanField(
+        default=False
+    )  # To check whether the user is banned
+    is_staff = models.BooleanField(default=False)
 
 
 class StudentProfile(models.Model):
-    pass
+    student = models.OneToOneField(User, on_delete=models.CASCADE, null=True)
+    division = models.CharField(max_length=1, default="A")
+    department = models.CharField(max_length=30, default=None)
+    admission_year = models.PositiveSmallIntegerField(default=1)
```

Comment:
Remove default.

Reply:
It says these are non nullable fields so I had to add default

Reply:
See `student` field (rename to `user` field) should not be null.
`division` default is okay.
`department` default is blank. Set blank=True.
`admission_year` default = 1 doesn't make sense. What is year = 1? Set it as null=True.